### PR TITLE
[Fix][useKeyboard] Adapt unsubscription for new EventEmitter API

### PR DESCRIPTION
**Pull request by retyui**
# Summary

Issue: https://github.com/react-native-community/hooks/pull/268#issuecomment-966604027

```jsx
// React Native 0.65.x +
typeof Keyboard.removeListener // function
Keyboard.removeListener(...) // print warning
```

### What's required for testing (prerequisites)?

- React Native >= `0.65`

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |
| Web     |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I've created a snack to demonstrate the changes: LINK HERE

### CODE REVIEW

File: src/useKeyboard.ts [hunk 1]
```diff
@@ -50,17 +50,17 @@ export function useKeyboard() {
     ]
 
     return () => {
-      if (Keyboard.removeListener) {
```
Comment (retyui) [inline src/useKeyboard.ts:53]: This check work until the RN team decided to restore this method: https://github.com/facebook/react-native/commit/035718ba97bb44c68f2a4ccdd95e537e3d28690c

File: src/useKeyboard.ts [hunk 2]
```diff
@@ -50,17 +50,17 @@ export function useKeyboard() {
...
+      if (typeof subscriptions?.[0]?.remove === 'function') {
```
Comment (LinusU) [inline src/useKeyboard.ts:53]: `subscriptions[0]` always exists here.

```suggestion
      if (typeof subscriptions[0].remove === 'function') {
```

**Review by LinusU — APPROVED** [review]
Neat, just one small nit

File: src/useKeyboard.ts (hunk 2 above)
Comment (retyui) [reply src/useKeyboard.ts:53]: I recheck it on latest versions

Comment (retyui) [reply src/useKeyboard.ts]: you right  RN [0.59.x...0.6x.x] always return EmitterSubscription with remove method, so this `if` block don't make sense anymore

https://github.com/facebook/react-native/blob/c20070f10458d48d6ac1eaac49e681e932bfb9fd/Libraries/EventEmitter/NativeEventEmitter.js#L41-L50

**Review by LinusU — APPROVED** [review]
Amazing! 🎉

**Comment by pvinis** [comment]
:rocket: PR was released in [`v2.8.1`](https://github.com/react-native-community/hooks/releases/tag/v2.8.1) :rocket:

**Comment by wildseansy** [comment]
Just to be clear, v2.8.1 won't support `<= v0.64` anymore, which uses `removeListener` - https://reactnative.dev/docs/0.64/keyboard

File: src/useKeyboard.ts (hunk 1 above)
Comment (wildseansy) [reply src/useKeyboard.ts:53]: My recommendation -- do a version check >= 0.65 -> use `.remove`. and < 0.65 -> use `.removeListener` to support all versions of react native

Comment (retyui) [reply src/useKeyboard.ts:53]: I checked  0.60.x , 0.61.x, 0.62.x  RN versions

and in these versions addEventListner returned subscription with remove method

**Comment by wildseansy** [comment]
Oh yay! thanks for checking

**Pull request merged**